File: WinForms/.NET Framework/SearchText/CS/AppForm.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using O2S.Components.PDFView4NET.Text;

namespace SearchText
{
    public partial class AppForm : Form
    {
        public AppForm()
        {
            InitializeComponent();
        }

        private bool incrementalSearchStarted = false;

        private void AppForm_Load(object sender, EventArgs e)
        {
            txtFile.Text = "..\\..\\..\\..\\..\\SupportFiles\\multicolumntextandimages.pdf";
            pdfDoc.FilePath = txtFile.Text;
        }

        private void btnBrowse_Click(object sender, EventArgs e)
        {
            if (ofd.ShowDialog() == DialogResult.OK)
            {
                txtFile.Text = ofd.FileName;
                pdfDoc.FilePath = txtFile.Text;
            }
        }

        private void btnFindAll_Click(object sender, EventArgs e)
        {
            incrementalSearchStarted = false;
            pageView.SearchText(txtSearchText.Text, PDFHighlightSearchResultsMode.AllResults);
        }

        private void btnFindNext_Click(object sender, EventArgs e)
        {
            if (!incrementalSearchStarted)
            {
                pageView.SearchText(txtSearchText.Text, PDFHighlightSearchResultsMode.SingleResult);
                incrementalSearchStarted = true;
            }
            else
            {
                pageView.HighlightNextSearchResult();
            }
        }

        private void btnClear_Click(object sender, EventArgs e)
        {
            pageView.ClearSearch();
            incrementalSearchStarted = false;
        }
    }
}

File: WinForms/.NET Framework/SearchText/VBNET/AppForm.vb
Imports O2S.Components.PDFView4NET
Imports O2S.Components.PDFView4NET.Text

Public Class AppForm
    Dim incrementalSearchStarted As Boolean = False

    Private Sub AppForm_Load(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles MyBase.Load
        txtFile.Text = "..\..\..\..\..\SupportFiles\\multicolumntextandimages.pdf"
        pdfDoc.FilePath = txtFile.Text
    End Sub

    Private Sub btnBrowse_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles btnBrowse.Click
        If (ofd.ShowDialog() = DialogResult.OK) Then
            txtFile.Text = ofd.FileName
            pdfDoc.FilePath = txtFile.Text
        End If
    End Sub

    Private Sub btnFindAll_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles btnFindAll.Click
        incrementalSearchStarted = False
        pageView.SearchText(txtSearchText.Text, PDFHighlightSearchResultsMode.AllResults)
    End Sub

    Private Sub btnFindNext_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles btnFindNext.Click
        If (Not incrementalSearchStarted) Then
            pageView.SearchText(txtSearchText.Text, PDFHighlightSearchResultsMode.SingleResult)
            incrementalSearchStarted = True
        Else
            pageView.HighlightNextSearchResult()
        End If
    End Sub

    Private Sub btnClear_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles btnClear.Click
        pageView.ClearSearch()
        incrementalSearchStarted = False
    End Sub
End Class
